utils/serialization/envelope_wrapper_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package serialization_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"github.com/hyperledger/fabric-x-common/protoutil"
+
+	"github.com/hyperledger/fabric-x-committer/utils/serialization"
+	"github.com/hyperledger/fabric-x-committer/utils/test"
+)
+
+// TestUnwrapEnvelopeBadInput tests UnwrapEnvelope function with invalid inputs.
+func TestUnwrapEnvelopeBadInput(t *testing.T) {
+	t.Parallel()
+	t.Run("Not an envelope", func(t *testing.T) {
+		t.Parallel()
+		_, _, err := serialization.UnwrapEnvelope([]byte("invalid input"))
+		require.Error(t, err)
+	})
+
+	t.Run("OK Header with an invalid payload", func(t *testing.T) {
+		t.Parallel()
+		envelope := &common.Envelope{
+			Payload: []byte("not-a-payload"),
+		}
+
+		envelopeBytes := protoutil.MarshalOrPanic(envelope)
```

Comment:
nit: it is preferred to avoid intermediate variables when possible
```suggestion
		envelopeBytes := protoutil.MarshalOrPanic(&common.Envelope{
			Payload: []byte("not-a-payload"),
		})
```